src/test/java/com/example/bookstore/service/shoppingcart/impl/ShoppingCartServiceImplTest.java

```diff
@@ -0,0 +1,187 @@
+            .setPassword("12345678")
+            .setRoles(Set.of(new Role(RoleName.ROLE_USER)))
+            .setShippingAddress("Kyiv, NP 3");
+    private static final Category VALID_CATEGORY = new Category()
+            .setId(VALID_ID)
+            .setName("Fantasy");
+
+    private static final Book VALID_BOOK = new Book()
+            .setId(VALID_ID)
+            .setTitle("Harry Potter")
+            .setAuthor("J. K. Rowling")
+            .setIsbn("921-3-ewfwefwef-148410-")
+            .setPrice(BigDecimal.valueOf(50.00))
+            .setCategories(Set.of(VALID_CATEGORY));
+    private static final CartItem VALID_CART_ITEM = new CartItem()
+            .setId(VALID_ID)
+            .setBook(VALID_BOOK)
+            .setQuantity(5);
+    private static final ShoppingCart VALID_SHOPPING_CART =
+            new ShoppingCart()
+                    .setId(VALID_ID)
+                    .setUser(VALID_USER)
+                    .setCartItems(Set.of(VALID_CART_ITEM));
+
+    private static final UpdateCartItemRequestDto VALID_UPDATE_ITEM_REQUEST =
+            new UpdateCartItemRequestDto()
+                    .setQuantity(5);
+    private static final AddCartItemRequestDto VALID_ADD_ITEM_REQUEST =
+            new AddCartItemRequestDto()
+                    .setBookId(VALID_ID)
+                    .setQuantity(5);
+
+    private static final CartItemDto CART_ITEM_DTO =
+            new CartItemDto()
+                    .setId(VALID_ID)
+                    .setTitle("Harry Potter")
+                    .setBookId(VALID_ID)
+                    .setQuantity(5);
+    private static final ShoppingCartDto SHOPPING_CART_DTO =
+            new ShoppingCartDto()
+                    .setId(VALID_ID)
+                    .setCartItems(Set.of(CART_ITEM_DTO))
+                    .setUserId(VALID_ID);
+    private static final Authentication AUTHENTICATION =
+            mock(Authentication.class);
+
+    static {
+        VALID_CART_ITEM.setShoppingCart(VALID_SHOPPING_CART);
+    }
+
+    @Mock
+    private ShoppingCartRepository shoppingCartRepository;
+    @Mock
+    private CartItemRepository cartItemRepository;
+    @Mock
+    private ShoppingCartMapper shoppingCartMapper;
+    @Mock
+    private CartItemMapper cartItemMapper;
+
+    @InjectMocks
+    private ShoppingCartServiceImpl shoppingCartService;
+
+    @Test
+    @DisplayName("Test retrieving an existing shopping cart for the user")
+    public void getShoppingCart_ExistingUser_ReturnsShoppingCart() {
+
+        when(AUTHENTICATION.getPrincipal()).thenReturn(VALID_USER);
+
+        when(shoppingCartRepository.findByUserId(VALID_USER.getId()))
+                .thenReturn(Optional.of(VALID_SHOPPING_CART));
+        when(shoppingCartMapper.toDto(VALID_SHOPPING_CART)).thenReturn(SHOPPING_CART_DTO);
+
+        ShoppingCartDto actual = shoppingCartService.getShoppingCart(AUTHENTICATION);
+
+        verify(shoppingCartRepository, times(1)).findByUserId(VALID_USER.getId());
```

Comment:
as far as I remember times(1) is default value, so it is not necessary to write it explicitly